lib/detect-libc.js

```diff
@@ -77,31 +79,61 @@ const familyFromCommand = (out) => {
  * @returns {Promise<?string>}
  */
 const family = async () => {
-  let family = null;
+  if (cachedFamily !== undefined) {
+    return cachedFamily;
+  }
+  cachedFamily = null;
   if (isLinux()) {
-    family = familyFromReport();
-    if (!family) {
-      const out = await safeCommand();
-      family = familyFromCommand(out);
+    try {
+      cachedFamily = await readFile(LDD_PATH)
+        .then(content => content.includes(MUSL))
+        .then(isMusl => isMusl ? MUSL : GLIBC);
+    } catch (e) {
+      // if permission model block access, we fallback to the old version
+      if (e.code === 'ERR_ACCESS_DENIED') {
+        cachedFamily = familyFromReport();
+        if (!cachedFamily) {
+          const out = await safeCommand();
+          cachedFamily = familyFromCommand(out);
+        }
+      } else {
+        // If ldd don't exist, accords https://wiki.musl-libc.org/faq.html we can assume
```

Comment:
I'm unsure we can assume musl in the absence of this shell script. glibc-based Linux will work quite happily without the `ldd` file as you can run `/lib64/ld-linux.so.2 ...` directly.

Reply:
Hm... so what file path do you recommend to check against? Or you prefer to run the other methods here?

Reply:
If the `/usr/bin/ldd` file is missing we must fallback to the existing methods.